### PR TITLE
feat: always return states in gcrf for trajectory models

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp [hunk 1]
```diff
@@ -104,35 +104,36 @@ State Tabulated::calculateStateAt(const Instant& anInstant) const
...
+                                    previousState.getVelocity().accessCoordinates()),
+                       previousState.getVelocity().getUnit(),
+                       previousState.getVelocity().accessFrame()
+                   },
+                   Quaternion::SLERP(previousState.getAttitude(), nextState.getAttitude(), ratio),
+                   previousState.getAngularVelocity() +
+                       ratio * (nextState.getAngularVelocity() - previousState.getAngularVelocity()),
+                   previousState.getFrame()
+        )
+            .inFrame(Frame::GCRF());
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp:128]:
```suggestion
        return State(
                   anInstant,
                   Position(
                       previousState.getPosition().accessCoordinates() +
                           ratio * (nextState.getPosition().accessCoordinates() -
                                    previousState.getPosition().accessCoordinates()),
                       previousState.getPosition().getUnit(),
                       previousState.getPosition().accessFrame()
                   ),
                   Velocity(
                       previousState.getVelocity().accessCoordinates() +
                           ratio * (nextState.getVelocity().accessCoordinates() -
                                    previousState.getVelocity().accessCoordinates()),
                       previousState.getVelocity().getUnit(),
                       previousState.getVelocity().accessFrame()
                   ),
                   Quaternion::SLERP(previousState.getAttitude(), nextState.getAttitude(), ratio),
                   Vector3d(previousState.getAngularVelocity() +
                       ratio * (nextState.getAngularVelocity() - previousState.getAngularVelocity())),
                   previousState.getFrame()
        )
            .inFrame(Frame::GCRF());
```

File: src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp [hunk 2]
```diff
@@ -104,35 +104,36 @@ State Tabulated::calculateStateAt(const Instant& anInstant) const
...
     }
     else if (stateRange.first != nullptr)
     {
-        return *(stateRange.first);
+        return (*(stateRange.first)).inFrame(Frame::GCRF());
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp:132]:
```suggestion
        return State(*(stateRange.first)).inFrame(Frame::GCRF());
```

File: src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp [hunk 3]
```diff
@@ -104,35 +104,36 @@ State Tabulated::calculateStateAt(const Instant& anInstant) const
...
     }
     else if (stateRange.second != nullptr)
     {
-        return *(stateRange.second);
+        return (*(stateRange.second)).inFrame(Frame::GCRF());
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp:136]:
```suggestion
        return State(*(stateRange.second)).inFrame(Frame::GCRF());
```

File: src/OpenSpaceToolkit/Astrodynamics/Trajectory/Model/Static.cpp [hunk 4]
```diff
@@ -18,6 +18,10 @@ Static::Static(const Position& aPosition)
     : Model(),
       position_(aPosition)
 {
+    if (aPosition.accessFrame() != Frame::ITRF())
+    {
+        throw ostk::core::error::runtime::Wrong("Position Frame", aPosition.accessFrame()->getName());
+    }
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Astrodynamics/Trajectory/Model/Static.cpp:24]: While you're at it, can you add a docstring to this class to indicate that this for static positions on the earth's surface (hence why the frame needs to be in ITRF), cause I never really knew what this class was for

https://github.com/open-space-collective/open-space-toolkit-astrodynamics/blob/main/include/OpenSpaceToolkit/Astrodynamics/Trajectory/Model/Static.hpp

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory.test.cpp
Comment (apaletta3) [inline test/OpenSpaceToolkit/Astrodynamics/Trajectory.test.cpp:1]: outside the scope of this MR but is there a way to enforce these style checks that we make everytime we touch an old file?
Or not cause its removing `usings` only?

**Review by apaletta3 — APPROVED** [review]
LGTM

File: src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp (hunk 2 above)
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp:132]: We don't need to create a new state, we're just converting the state that this is pointing to and converting that and returning. Seems unnecessary.

File: src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp (hunk 3 above)
Comment (vishwa2710) [reply src/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.cpp:136]: Idem

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory.test.cpp
Comment (vishwa2710) [reply test/OpenSpaceToolkit/Astrodynamics/Trajectory.test.cpp:1]: This was just me moving around the usings. At some point we should just apply across the repo but I update it when I see it file by file to reduce the overall effort when we eventually do it.

**Pull request merged**